extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts

```diff
@@ -197,88 +197,104 @@ export class ResourceTypePickerDialog extends DialogBase {
 	private updateToolsDisplayTable(): void {
 		this.toolRefreshTimestamp = new Date().getTime();
 		const currentRefreshTimestamp = this.toolRefreshTimestamp;
-		const toolRequirements = this.getCurrentProvider().requiredTools;
 		const headerRowHeight = 28;
-		this._toolsTable.height = 25 * Math.max(toolRequirements.length, 1) + headerRowHeight;
+		this._toolsTable.height = 25 * Math.max(this.toolRequirements.length, 1) + headerRowHeight;
 		this._dialogObject.message = {
 			text: ''
 		};
 		this._installToolButton.hidden = true;
-		if (toolRequirements.length === 0) {
+		if (this.toolRequirements.length === 0) {
 			this._dialogObject.okButton.enabled = true;
 			this._toolsTable.data = [[localize('deploymentDialog.NoRequiredTool', "No tools required"), '']];
 			this._tools = [];
 		} else {
-			this._tools = toolRequirements.map(toolReq => {
+			this._tools = this.toolRequirements.map(toolReq => {
 				return this.toolsService.getToolByName(toolReq.name)!;
 			});
 			this._toolsLoadingComponent.loading = true;
 			this._dialogObject.okButton.enabled = false;
-			Promise.all(
-				this._tools.map(tool => tool.loadInformation())
-			).then(async () => {
-				// If the local timestamp does not match the class level timestamp, it means user has changed options, ignore the results
-				if (this.toolRefreshTimestamp !== currentRefreshTimestamp) {
-					return;
-				}
-				let minVersionCheckFailed = false;
-				const toolsToAutoInstall: ITool[] = [];
-				let messages: string[] = [];
-				this._toolsTable.data = toolRequirements.map(toolRequirement => {
-					const tool = this.toolsService.getToolByName(toolRequirement.name)!;
-					// subscribe to onUpdateData event of the tool.
-					this._toDispose.push(tool.onDidUpdateData((t: ITool) => {
-						this.updateToolsDisplayTableData(t);
-					}));
-					if (tool.isNotInstalled) {
-						if (tool.autoInstallSupported) {
-							toolsToAutoInstall.push(tool);
-						} else {
-							messages.push(localize('deploymentDialog.ToolInformation', "'{0}' was not discovered and automated installation is not currently supported. Install '{0}' manually or ensure it is started and discoverable. Once done please restart Azure Data Studio. See [{1}] .", tool.displayName, tool.homePage));
-						}
-					} else if (tool.isInstalled && toolRequirement.version && !tool.isSameOrNewerThan(toolRequirement.version)) {
-						minVersionCheckFailed = true;
-						messages.push(localize('deploymentDialog.ToolDoesNotMeetVersionRequirement', "'{0}' [ {1} ] does not meet the minimum version requirement, please uninstall it and restart Azure Data Studio.", tool.displayName, tool.homePage));
+			Promise.all(this._tools.map(tool => tool.loadInformation()))
```

Comment:
1. You still have a dangling promise here from the Promise.all
2. If there are multiple errors you'll only ever display one at a time - overwriting the previous error if a new one occurs

What seems to make more sense for this is to either :

1. As the errors occur append them to the message instead of setting it to just that error each time
2. Capture all the errors and only at the end display all the ones that occurred

Reply:
@ranasaria - I assume your response was meant for this. The issue here is you're having a catch for every promise within the Promise.all - which means that even if an error happens in one the rest will still continue to execute and if an error occurs in those they'll also set the error message (and thus overwriting the previous one). 

Having the catch on each internal promise is fine - but my thought is that you should just be capturing the list of errors that occur and either appending the error message or just waiting until they're all done then setting the error message with all the errors concatenated together.

Reply:
(and if you do this you should also have a catch on the Promise.all. Technically it is pretty much impossible for an error to occur but the linter doesn't know that and it's just good practice to be 100% sure anyways. For that one you can just call onUnexpectedError I'd say)

Reply:
Will do.